src/Raven.Server/Web/System/TcpConnectionInfoHandler.cs

```diff
@@ -162,12 +163,19 @@ private List<string> GetResponsibleNodes(DatabaseTopology topology, string datab
             var mentorNodeTask = new PullNodeTask
             {
                 Mentor = mentorNode,
+                PinToMentorNode = pinToMentorNode,
                 DatabaseGroupId = databaseGroupId
             };
 
             while (topology.Members.Count > 0)
             {
                 var next = topology.WhoseTaskIsIt(ServerStore.CurrentRachisState, mentorNodeTask, null);
+                if (next.Equals(mentorNode) && pinToMentorNode)
+                {
+                    // if it's pinned to the mentor node than the list will always contain one node (only if the node exists in the topology)
+                    list.Add(next);
+                    return list;
+                }
```

Comment:
This is really strange, why do that in the while?
Check `pinToMentorNode` and see if the mentor node is in the members, then return it directly, no?

Reply:
We break the behavior of the getting adding and removing the nodes to the list.
Let's say I'm using -
```
if(pinToMentor)
{
  next = WoseTaskIsit
  list.Add()
}
else{
while(Memebers.Count > 0)
add and remove in order to get the order
}
```
In any case, if we remove the node from the cluster it will do failover.